geopandas/geoseries.py

```diff
@@ -170,7 +168,10 @@ def _wrapped_pandas_method(self, mtd, *args, **kwargs):
 
     def __getitem__(self, key):
         if isinstance(key, (slice, list, Series, np.ndarray)):
-            return super(GeoSeries, self).__getitem__(key)
+            block = self._data._block[key]
+            index = self.index[key]
+            return GeoSeries(SingleBlockManager(block, axis=index),
```

Comment:
So slicing a GeoDataFrame works nicely (at least a DataFrame with GeometryBlock), due to the `_slice` method implemented on GeometryBlock.

However, when slicing a Series, the SingleBlockManager.get_slice method is used: https://github.com/pandas-dev/pandas/blob/793020293ee1e5fa023f45c12943a4ac51cc23d0/pandas/core/internals.py#L4273-L4278

This one seems to use the block `_slice` method in a not fully correct way -> it slices it, but then generally creates a new SingleBlockManager (`__class__(..)` from the slices values (not block anymore, but sliced underlying values), instead of preserving the block type, which it easily could I think.

So this is something I could solve on the pandas side, but of course easier if for now we have a fix here as well.